modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -927,6 +911,68 @@ public int getSubscriptionCount( String queueName){
     }
 
     /**
+     * Method to construct message body of a single message.
+     *
+     * @param bodySize   original content size of the message
+     * @param amqMessage AMQMessage
+     * @return message content as bytes and last position of written data as an int value
+     * @throws MBeanException
+     */
+    private HashMap<List<Byte>, Integer> constructContent(int bodySize, AMQMessage amqMessage) throws MBeanException {
+
+        List<Byte> messageContent = new ArrayList<Byte>();
+        java.nio.ByteBuffer buffer = java.nio.ByteBuffer.allocate(bodySize);
+        int position = 0;
+
+        if (ContentCompressionDecompressionHandler.isCompressionEnabled
+                && (bodySize > ContentCompressionDecompressionHandler.contentCompressionThreshold)) {
+            /* If the current message was compressed by the server (if compression is enabled and content size
+             * of the message is larger than the content compression threshold), decompress the message content
+             * and, put decompressed data into the array list
+             */
+            Set<Long> messageToFetch = new HashSet<>();
+            Long messageID = amqMessage.getMessageId();
+            messageToFetch.add(messageID);
+
+            try {
+                Map<Long, List<AndesMessagePart>> contentListMap = MessagingEngine.getInstance()
+                        .getContent(new ArrayList<>(messageToFetch));
+                List<AndesMessagePart> contentList = contentListMap.get(messageID);
+
+                byte[] decompressedMessage = ContentCompressionDecompressionHandler.getInstance()
+                        .getDecompressedMessage(contentList, bodySize);
+
+                messageContent.clear();
+                while (position < decompressedMessage.length) {
+                    messageContent.add(position, decompressedMessage[position]);
+                    position++;
+                }
+                position = position - 1;
+            } catch (AndesException e) {
+                throw new MBeanException(e, "Error occurred while construct message content. Message ID:"
+                        + amqMessage.getMessageId());
+            }
+        } else {
+            while (position < bodySize) {
+                position += amqMessage.getContent(buffer, position);
```

Comment:
This is more readable if you change this as position = position + amqMessage.getContent(buffer, position);